models/models.go

```diff
@@ -12,20 +12,27 @@ type User struct {
 }
 
 type Document struct {
-	ID           uint `gorm:"primaryKey"`
-	AuthorID     uint
-	Title        string
-	Content      string
-	Appendix     string
-	Status       string `gorm:"not null;default:'EDIT'"` // EDIT, VERIFY, REJECT, APPROVE
-	RejectReason string
+	ID       uint `gorm:"primaryKey"`
+	AuthorID uint
+	Title    string
+	Content  string
+	Appendix string
+	Status   string `gorm:"not null;default:'EDIT'"` // EDIT, VERIFY, REJECT, APPROVE
+	// RejectReason string // should be in history table
 	CreatedAt    time.Time // 建立時間（由GORM自動管理）
 	UpdatedAt    time.Time // 最後一次更新時間（由GORM自動管理）
 	ApprovedDate time.Time
 	// foreign keys to User
 	ApproverID uint `gorm:"foreignKey:ApproverID"`
-	ViewerID   uint `gorm:"foreignKey:ViewerID"`
 	Author     User `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
 	Approver   User `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
-	Viewer     User `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
+}
+
+type DocumentViewer struct {
+	ID uint `gorm:"primaryKey"`
+	// foreign keys to Document and User
+	DocumentID uint     `gorm:"foreignKey:DocumentID"`
+	ViewerID   uint     `gorm:"foreignKey:ViewerID"`
+	Document   Document `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
+	Viewer     User     `gorm:"constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
```

Comment:
Love the way you define foreign key constraints. I'll modify the _documents_ and _users_ tables to use this approach for the author ID.